### PR TITLE
Structure_Engine: Add default descriptions to missing property objects

### LINKED ISSUE #1636: Structure_Engine: Add Description method for all IProperty objects
#### Description:

All objects implementing the `IProperty` interface should have a `Description()` method.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #1636 

Adding descriptions to missing property-type objects. Aim for methods is to be used as default names in adapter when the name has not been set.

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EtE9dAU1IsVDl72FMOeenHoBw6w1G8AHWsjCzOyN3G3yww?e=pYgZEJ

### CODE REVIEW

File: Structure_Engine/Query/DescriptionOrName.cs [hunk 1]
```diff
@@ -42,6 +43,29 @@ public static partial class Query
         /**** Public Methods                            ****/
         /***************************************************/
 
+        [Description("Gets the name from a section. If null or empty, a default description name is provided instead.")]
+        [Input("section", "The section go get the name or default description from.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/DescriptionOrName.cs]: "The section to get the name or default description from."
"The section whose name we desire" - OK that's a little personal for a software help message

File: Structure_Engine/Query/Description.cs [hunk 2]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
             return "LoadingPanel Application: " + property.LoadApplication + " RefEdge: " + property.ReferenceEdge;
         }
 
+        /***************************************************/
+        /**** Public Methods - Constraints              ****/
+        /***************************************************/
+
+        [Description("Generates a default description for the Constraint3DOF as 'DOftypes-Values'.")]
+        [Input("constraint", "The Constraint3DOF to get a default description for.")]
+        [Output("desc", "The generated descritpion for the constraint.")]
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_Engine/Query/Description.cs:308]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 3]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+                desc += "-" + constraint.KX + "," + constraint.KY + "," + constraint.KNorm; 
+            }
+            return desc;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the Constraint4DOF as 'DOftypes-Values'.")]
+        [Input("constraint", "The Constraint4DOF to get a default description for.")]
+        [Output("desc", "The generated descritpion for the constraint.")]
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_Engine/Query/Description.cs:323]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 4]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+                        "," + constraint.RotationalStiffnessX;
+            }
+            return desc;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the Constraint6DOF as 'DOftypes-Values'.")]
+        [Input("constraint", "The Constraint6DOF to get a default description for.")]
+        [Output("desc", "The generated descritpion for the constraint.")]
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_Engine/Query/Description.cs:341]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 5]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+                        "," + constraint.RotationalStiffnessX +  "," + constraint.RotationalStiffnessY +  "," + constraint.RotationalStiffnessZ;
+            }
+            return desc;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the BarRelease as 'StartReleaseNameOrDesc - EndReleaseNameOrDesc'.")]
+        [Input("release", "The Constraint6DOF to get a default description for.")]
+        [Output("desc", "The generated descritpion for the constraint.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:359]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 6]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+            return release.StartRelease.DescriptionOrName() + " - " + release.EndRelease.DescriptionOrName();
+        }
+
+        /***************************************************/
+        /**** Public Methods - Offset                   ****/
+        /***************************************************/
+
+        [Description("Generates a default description for the Offset as 'Start - End'.")]
+        [Input("offset", "The Constraint6DOF to get a default description for.")]
+        [Output("desc", "The generated descritpion for the constraint.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:371]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 7]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+            return offset.Start.VectorComponents() + " - " + offset.End.VectorComponents();
+        }
+
+        /***************************************************/
+        /**** Public Methods - Material                 ****/
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_Engine/Query/Description.cs:383]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 8]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this Steel material)
+        {
+            return "Steel " + material.MaterialAnalyticalValues() + ", fy:" + material.YieldStress + ", fu:" + material.UltimateStress;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:393]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 9]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this Concrete material)
+        {
+            return "Concrete " + material.MaterialAnalyticalValues() + ", cyl:" + material.CylinderStrength + ", cube:" + material.CubeStrength;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:403]:
```suggestion
        [Output("desc", "The generated description for the constraint.")]
```

File: Structure_Engine/Query/Description.cs [hunk 10]
```diff
@@ -118,7 +122,7 @@ public static string Description(this GeneralisedFabricatedBoxProfile profile)
         [Output("desc", "The generated descritpion for the profile depending on its dimensions.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:122]:
```suggestion
        [Output("desc", "The generated description for the profile depending on its dimensions.")]
```

File: Structure_Engine/Query/Description.cs [hunk 11]
```diff
@@ -262,7 +266,7 @@ public static string Description(this ExplicitSection section)
         [Output("desc", "The generated descritpion for the property depending on its dimensions, material and type.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:266]:
```suggestion
        [Output("desc", "The generated description for the property depending on its dimensions, material and type.")]
```

File: Structure_Engine/Query/Description.cs [hunk 12]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this Aluminium material)
+        {
+            return "Alum " + material.MaterialAnalyticalValues();
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:413]:
```suggestion
        [Output("desc", "The generated description for the material.")]
```

File: Structure_Engine/Query/Description.cs [hunk 13]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this GenericIsotropicMaterial material)
+        {
+            return "Gen " + material.MaterialAnalyticalValues();
+        }
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:421]:
```suggestion
        [Output("desc", "The generated description for the material.")]
```

File: Structure_Engine/Query/Description.cs [hunk 14]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this Timber material)
+        {
+            return "Timber " + material.MaterialAnalyticalValues();
+        }
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:429]:
```suggestion
        [Output("desc", "The generated description for the material.")]
```

File: Structure_Engine/Query/Description.cs [hunk 15]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+            return "Gen " + material.MaterialAnalyticalValues();
+        }
+
         /***************************************************/
         /**** Public Methods - Interfaces               ****/
         /***************************************************/
 
+        [Description("Generates a default description for the Iproeprty, depending on type and its properties.")]
+        [Input("property", "The property to get a default description for.")]
+        [Output("desc", "The generated descritpion for the property, depending on its proeprty values.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:441]:
```suggestion
        [Output("desc", "The generated description for the property, depending on its property values.")]
```

File: Structure_Engine/Query/Description.cs [hunk 16]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string IDescription(this IProperty property)
+        {
+            return Description(property as dynamic);
+        }
+
+        /***************************************************/
+
         [Description("Generates a default description for the Section, depending on type, profile and material.")]
         [Input("section", "The section to get a default description for.")]
         [Output("desc", "The generated descritpion for the section depending on its dimensions, material and type.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:451]:
```suggestion
        [Output("desc", "The generated description for the section depending on its dimensions, material and type.")]
```

File: Structure_Engine/Query/Description.cs [hunk 17]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
+        public static string Description(this Steel material)
+        {
+            return "Steel " + material.MaterialAnalyticalValues() + ", fy:" + material.YieldStress + ", fu:" + material.UltimateStress;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_Engine/Query/Description.cs:391]:
```suggestion
        [Description("Generates a default description for the material based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 18]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this Offset offset)
+        {
+            return offset.Start.VectorComponents() + " - " + offset.End.VectorComponents();
+        }
+
+        /***************************************************/
+        /**** Public Methods - Material                 ****/
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:381]:
```suggestion
        [Description("Generates a default description for the material based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 19]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
+        public static string Description(this Concrete material)
+        {
+            return "Concrete " + material.MaterialAnalyticalValues() + ", cyl:" + material.CylinderStrength + ", cube:" + material.CubeStrength;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:401]:
```suggestion
        [Description("Generates a default description for the material based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 20]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
+        public static string Description(this Aluminium material)
+        {
+            return "Alum " + material.MaterialAnalyticalValues();
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:411]:
```suggestion
        [Description("Generates a default description for the material based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 21]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
+        public static string Description(this GenericIsotropicMaterial material)
+        {
+            return "Gen " + material.MaterialAnalyticalValues();
+        }
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:419]:
```suggestion
        [Description("Generates a default description for the material based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 22]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        [Input("material", "The material to get a default description for.")]
+        [Output("desc", "The generated descritpion for the material.")]
+        public static string Description(this Timber material)
+        {
+            return "Timber " + material.MaterialAnalyticalValues();
+        }
+
+        [Description("Generates a default description for the material as a description fo its properties.")]
```
Comment (JosefTaylor) [inline Structure_Engine/Query/Description.cs:427]:
```suggestion
        [Description("Generates a default description for the material based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 23]
```diff
@@ -295,10 +299,153 @@ public static string Description(this LoadingPanelProperty property)
...
+        public static string Description(this GenericOrthotropicMaterial material)
+        {
+            return "Gen " + material.MaterialAnalyticalValues();
+        }
+
         /***************************************************/
         /**** Public Methods - Interfaces               ****/
         /***************************************************/
 
+        [Description("Generates a default description for the Iproeprty, depending on type and its properties.")]
```
[the remark this answers is not in the document]
Comment (JosefTaylor) [reply Structure_Engine/Query/Description.cs:439]:
```suggestion
        [Description("Generates a default description for the IProperty, based on its properties.")]
```

File: Structure_Engine/Query/Description.cs [hunk 24]
```diff
@@ -115,10 +119,10 @@ public static string Description(this GeneralisedFabricatedBoxProfile profile)
 
         [Description("Generates a default description for the profile as 'Kite Angle x Width x Thickness'.")]
         [Input("profile", "The profile to get a default description for.")]
-        [Output("desc", "The generated descritpion for the profile depending on its dimensions.")]
+        [Output("desc", "The generated description for the profile depending on its dimensions.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:122]: based on its dimensions instead of depending on

File: Structure_Engine/Query/Description.cs [hunk 25]
```diff
@@ -259,10 +263,10 @@ public static string Description(this ExplicitSection section)
 
         [Description("Generates a default description for the SurfaceProperty as 'THK Thickness - MaterialName'.")]
         [Input("property", "The SurfaceProperty to get a default description for.")]
-        [Output("desc", "The generated descritpion for the property depending on its dimensions, material and type.")]
+        [Output("desc", "The generated description for the property depending on its dimensions, material and type.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:266]: based on its dimensions instead of depending on

File: Structure_Engine/Query/Description.cs [hunk 26]
```diff
@@ -295,13 +299,156 @@ public static string Description(this LoadingPanelProperty property)
             return "LoadingPanel Application: " + property.LoadApplication + " RefEdge: " + property.ReferenceEdge;
         }
 
+        /***************************************************/
+        /**** Public Methods - Constraints              ****/
+        /***************************************************/
+
+        [Description("Generates a default description for the Constraint3DOF as 'DOftypes-Values'.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:306]: DOFType values

File: Structure_Engine/Query/Description.cs [hunk 27]
```diff
@@ -295,13 +299,156 @@ public static string Description(this LoadingPanelProperty property)
...
+            if (constraint.ElasticValuesWritten())
+            {
+                desc += "-" + constraint.KX + "," + constraint.KY + "," + constraint.KNorm; 
+            }
+            return desc;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the Constraint4DOF as 'DOftypes-Values'.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:321]: DOFType values

File: Structure_Engine/Query/Description.cs [hunk 28]
```diff
@@ -295,13 +299,156 @@ public static string Description(this LoadingPanelProperty property)
...
+            {
+                desc += "-" + constraint.TranslationalStiffnessX + "," + constraint.TranslationalStiffnessY + "," + constraint.TranslationalStiffnessZ +
+                        "," + constraint.RotationalStiffnessX;
+            }
+            return desc;
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the Constraint6DOF as 'DOftypes-Values'.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:339]: DOFType values

File: Structure_Engine/Query/Description.cs [hunk 29]
```diff
@@ -351,6 +498,202 @@ private static string CheckGetMaterialName(IMaterialFragment material)
...
+                case DOFType.Gap:
+                    return "g";
+                default:
+                    return "-";
+            }
+        }
+
+        /***************************************************/
+
+        [Description("Returns true if any of the dof elements is of a value based type or if any of the values are non-zero.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:571]: Returns true if any of the degrees of freedom is of a value based type or the values are non-zero.

File: Structure_Engine/Query/Description.cs [hunk 30]
```diff
@@ -351,6 +498,202 @@ private static string CheckGetMaterialName(IMaterialFragment material)
...
+
+            if (constraint.KNorm != 0)
+                return true;
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Returns true if any of the dof elements is of a value based type or if any of the values are non-zero.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:597]: Returns true if any of the degrees of freedom is of a value based type or the values are non-zero.

File: Structure_Engine/Query/Description.cs [hunk 31]
```diff
@@ -351,6 +498,202 @@ private static string CheckGetMaterialName(IMaterialFragment material)
...
+
+            if (constraint.RotationalStiffnessX != 0)
+                return true;
+
+            return false;
+        }
+
+        /***************************************************/
+
+        [Description("Returns true if any of the dof elements is of a value based type or if any of the values are non-zero.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:629]: Returns true if any of the degrees of freedom is of a value based type or the values are non-zero.

File: Structure_Engine/Query/Description.cs (hunk 26 above)
Comment (IsakNaslundBh) [reply Structure_Engine/Query/Description.cs:306]: not sure on making this change. Intention was " 'DOF type description' - 'Any numerical values'"

Comment (peterjamesnugent) [reply Structure_Engine/Query/Description.cs:306]: I think it's a bit too wordy, could we not make it simpler? 

Generates a default description for the Constraint3DOF based on the constraints at each degree of freedom.

Comment (IsakNaslundBh) [reply Structure_Engine/Query/Description.cs:306]: happy with that.

File: Structure_Engine/Query/Description.cs [hunk 32]
```diff
@@ -289,39 +293,182 @@ public static string Description(this Waffle property)
...
+        [Output("desc", "The generated description for the profile based on its dimensions.")]
         public static string IDescription(this IProfile profile)
         {
             return Description(profile as dynamic);
         }
 
         /***************************************************/
 
-        [Description("Generates a default description for the Section, depending on type, dimensions and material.")]
+        [Description("Generates a default description for the Section, based on type, dimensions and material.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:469]: for the SurfaceProperty

File: Structure_Engine/Query/Description.cs [hunk 33]
```diff
@@ -289,39 +293,182 @@ public static string Description(this Waffle property)
...
+        {
+            return Description(property as dynamic);
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the Section, based on type, profile and material.")]
         [Input("section", "The section to get a default description for.")]
-        [Output("desc", "The generated descritpion for the section depending on its dimensions, material and type.")]
+        [Output("desc", "The generated description for the section based on its dimensions, material and type.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:451]: SectionProperty

File: Structure_Engine/Query/Description.cs [hunk 34]
```diff
@@ -289,39 +293,182 @@ public static string Description(this Waffle property)
...
+        [Input("property", "The property to get a default description for.")]
+        [Output("desc", "The generated description for the property, based on its property values.")]
+        public static string IDescription(this IProperty property)
+        {
+            return Description(property as dynamic);
+        }
+
+        /***************************************************/
+
+        [Description("Generates a default description for the Section, based on type, profile and material.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:449]: SectionProperty

File: Structure_Engine/Query/Description.cs [hunk 35]
```diff
@@ -289,39 +293,182 @@ public static string Description(this Waffle property)
...
             return Description(section as dynamic);
         }
 
         /***************************************************/
 
-        [Description("Generates a default description for the Profile, depending on dimensions.")]
+        [Description("Generates a default description for the Profile, based on dimensions.")]
         [Input("profile", "The profile to get a default description for.")]
-        [Output("desc", "The generated descritpion for the profile depending on its dimensions.")]
+        [Output("desc", "The generated description for the profile based on its dimensions.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:461]: Profile

File: Structure_Engine/Query/Description.cs [hunk 36]
```diff
@@ -289,39 +293,182 @@ public static string Description(this Waffle property)
...
             return Description(profile as dynamic);
         }
 
         /***************************************************/
 
-        [Description("Generates a default description for the Section, depending on type, dimensions and material.")]
+        [Description("Generates a default description for the Section, based on type, dimensions and material.")]
         [Input("property", "The SurfaceProperty to get a default description for.")]
-        [Output("desc", "The generated descritpion for the property depending on its dimensions, material and type.")]
+        [Output("desc", "The generated description for the property based on its dimensions, material and type.")]
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/Description.cs:471]: SurfaceProperty

File: Structure_Engine/Query/IsNumericallyDependant.cs [hunk 37]
```diff
@@ -0,0 +1,167 @@
...
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+
+        [Description("Returns true if any of the degrees of freedom is of a value based type or the values are non-zero.")]
+        public static bool IsNumericallyDependant(this Constraint3DOF constraint)
```
Comment (peterjamesnugent) [inline Structure_Engine/Query/IsNumericallyDependant.cs:40]: IsNumericallyDependent 

The file name will have to change as well
